databases/week1/hyf-week1.sql

```diff
@@ -0,0 +1,51 @@
+USE `hyf-week1`;
+ 
+SELECT * FROM `task`;
+SELECT * FROM `user`;
+SELECT * FROM `status`;
+SELECT `status`. `name` FROM `task`;
+SELECT * FROM `task` WHERE status_id != 3;
+-- Find out how many tasks are in the task table
+SELECT COUNT(id) FROM `task`;
+
+-- Find out how many tasks in the task table do not have a valid due date
+SELECT * FROM `task`WHERE due_date IS NULL;
+
+-- Find all the tasks that are marked as done
+SELECT * FROM `task` WHERE status_id = 3;
+
+-- Find all the tasks that are not marked as done
+SELECT * FROM `task` WHERE status_id < 3;
+
+-- Get all the tasks, sorted with the most recently created first
+SELECT * FROM `task` ORDER BY due_date DESC;
+
+-- Get the single most recently created task
+SELECT * FROM `task` ORDER BY due_date DESC LIMIT 1;
+
+-- Get the title and due date of all tasks where the title or description contains database
+SELECT `title`, `due_date`
+FROM `task`
+WHERE `title` 
+LIKE '%database%' 
+OR `description` 
+LIKE '%database%';
```

Comment:
Also, look at 'REGEXP' as a more powerful replacement for 'LIKE'.
For example: 
SELECT `id`, `title`, `due_date`
FROM `task`
WHERE `title` REGEXP 'database' OR 
      `description` REGEXP 'database';